### DISCUSSION TITLE
Adding some parameters for better consumption control

### DISCUSSION THREAD

**Discussion by Elsbett1** [Hyundai-Kia-Connect/hyundai_kia_connect_api#621] [Ideas]
With the all new Hyundai Kona SX2 EV the consumption values are strange and confusing when calculated out of the different consumption values (engine_consumption, climate_consumption, onboard_electronics_consumption, regenerated_energy).

Would it be possible to add the following parameters in the API, to get a better control of consumption:

- ev_battery_capacity 
- ev_battery_remain

=>the unit of these values are strange (its not kWh), but if we set ev_battery_capacity with the net battery capacity in kWh, they are very useful.

Additional, it  could be helpful also to add the three consumption values that are displayed in the car (and which are quite reasonable):

- Drive (the consumption of the last ride in kWh/100km)
- AfterRefuel (the consumption since the last charging in kWh/100km)
- Accumulated (the consumption since the last manual reset of counter in kWh/100km)

=>maybe these values are also interesting for Hybrid or combustion engine, but with other unit?

I could verify with the debug.py script of ZuinigeRijder that all of the requested values are available in the Hyundai Server-Data.

Thanks for thinking about.

**Comment by cdnninja** [comment]
Open to them being added but need someone who has one of these cars to do so.

**Reply by Elsbett1** [reply]
I have such a car, how can I help? Would you need a debug log (made with ZuinigeRijders debug.py) where you see the needed parameters?
